### PR TITLE
Replace group with selector in SecurityPolicy

### PR DESCRIPTION
**Pull request by zwtop**
* Update SecurityPolicy types to follow Kubernetes NetworkPolicy mode.

* Update webhook to validate the new SecurityPolicy.

* Update policy controller support to generate groups.

* Update unit tests and e2e tests support the new SecurityPolicy.

#131

### CODE REVIEW

File: pkg/controller/policy/policy_controller.go [hunk 1]
```diff
@@ -522,45 +557,44 @@ func (r *PolicyReconciler) completePolicy(policy *securityv1alpha1.SecurityPolic
...
+				return nil, nil, err
+			}
+			for _, ipNet := range ipNets {
+				ipBlocks[ipNet.String()]++
+			}
+		default:
+			group := peerAsEndpointGroup(namespace, peer).GetName()
+			revision, ipAddrs, exist := r.groupCache.ListGroupIPBlocks(group)
+			if !exist {
+				return nil, nil, groupNotFound(fmt.Errorf("group %s members not found", group))
```
Comment (KeyongSun) [inline pkg/controller/policy/policy_controller.go:591]: The default case is EndpointSelector or NamespaceSelector, it's better to add a log if not exist for specific Selector for debug purpose.

File: pkg/apis/security/v1alpha1/types.go [hunk 2]
```diff
@@ -29,105 +29,145 @@ import (
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
 
+// SecurityPolicy describes what network traffic is allowed for a set of Endpoint.
+// Folow NetworkPolicy https://github.com/kubernetes/api/blob/v0.22.1/networking/v1/types.go#L29.
```
[the remark this answers is not in the document]
Comment (zwtop) [reply pkg/apis/security/v1alpha1/types.go:33]: done

File: pkg/apis/security/v1alpha1/types.go [hunk 3]
```diff
@@ -29,105 +29,145 @@ import (
...
-	// then this SecurityPolicy does not allow any traffic.
+	// List of ingress rules to be applied to the selected endpoints. If this field
+	// is empty then this SecurityPolicy does not allow any traffic.
+	// +optional
 	IngressRules []Rule `json:"ingressRules,omitempty"`
 
-	// List of egress rules to be applied to giving groups. If this field is empty
-	// then this SecurityPolicy limits all outgoing traffic.
+	// List of egress rules to be applied to the selected endpoints. If this field
+	// is empty then this NetworkPolicy limits all outgoing traffic.
```
[the remark this answers is not in the document]
Comment (zwtop) [reply pkg/apis/security/v1alpha1/types.go:61]: done

File: pkg/controller/policy/group_generator.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+	}
+}
+
+func (r *PolicyReconciler) getEndpointGroupFromSecurityPolicy(policy *securityv1alpha1.SecurityPolicy, groupName string) *groupv1alpha1.EndpointGroup {
+	for _, appliedTo := range policy.Spec.AppliedTo {
+		group := appliedAsEndpointGroup(policy.GetNamespace(), appliedTo)
+		if group != nil && group.GetName() == groupName {
+			return group
+		}
+	}
```
[the remark this answers is not in the document]
Comment (zwtop) [reply pkg/controller/policy/group_generator.go:135]: If the group name is the same, then the group is the same group.
So if the AppliedTo contains the group, just return it.

File: pkg/controller/policy/policy_controller.go (hunk 1 above)
Comment (zwtop) [reply pkg/controller/policy/policy_controller.go:591]: done

File: deploy/crds/security.lynx.smartx.com_securitypolicies.yaml [hunk 5]
```diff
@@ -19,6 +19,8 @@ spec:
   - name: v1alpha1
     schema:
       openAPIV3Schema:
+        description: SecurityPolicy describes what network traffic is allowed for
+          a set of Endpoint. Folow NetworkPolicy https://github.com/kubernetes/api/blob/v0.22.1/networking/v1/types.go#L29.
```
[the remark this answers is not in the document]
Comment (zwtop) [reply deploy/crds/security.lynx.smartx.com_securitypolicies.yaml:23]: done

File: pkg/webhook/validates/crd_validate.go [hunk 6]
```diff
@@ -206,7 +205,7 @@ func (v *CRDValidate) Validate(ar *admv1.AdmissionReview) *admv1.AdmissionRespon
 	}
 
 	if !allowed && msg != "" {
-		klog.Errorf("unallow to %s objects %s: %s", operation, gvk, msg)
+		klog.Errorf("unallow to %s objects %s %+v: %s", operation, gvk, namenamespace, msg)
```
[the remark this answers is not in the document]
Comment (zwtop) [reply pkg/webhook/validates/crd_validate.go:208]: done

**Comment by zwtop** [comment]
rebase origin/main

**Review by KeyongSun — APPROVED** [review]

**Pull request merged**